### ISSUE TITLE
ignoreExt directive seems to be ignored (Ubuntu 20.04, checkrr 1.2.0)

### ISSUE THREAD

**Issue by Emstar** [aetaric/checkrr#16]
The directive to ignore file extensions doesn't seem to work. Listed extensions are still scanned, and show up in the file count and log. 

Ignoring hidden files works as expected: a file like .test does not add to the count of files checked and does not appear in the logs at all. 

**Config file:**

```
checkpath:
- "/home/username/testrr"
lidarraddress: ""
lidarrapikey: ""
lidarrbaseurl: /
lidarrport: 8686
processlidarr: false
processradarr: false
processsonarr: false
radarraddress: ""
radarrapikey: ""
radarrbaseurl: /
radarrport: 7878
removeunknownfiles: false
sonarraddress: 127.0.0.1
sonarrapikey: ""
sonarrbaseurl: /
sonarrport: 8989
database: /var/lib/checkrr/checkrr.db
csvfile: /var/lib/checkrr/badfiles.csv
ignoreHidden: true
ignoreexts:
  - .txt
  - .nfo
  - .DS_Store
  - .csv
```

**Contents of test directory**

```
drwxrwxr-x  2 username username    4096 Oct 28 10:37  ./
drwxr-xr-x 20 username username    4096 Oct 28 09:52  ../
-rw-r--r--  1 username username 8657959 Oct 28 10:03 "test.mp3"
-rw-rw-r--  1 username username       7 Oct 28 10:33  .DS_Store
-rw-rw-r--  1 username username       4 Oct 28 10:37  .test
-rw-rw-r--  1 username username       7 Oct 28 10:00  test.csv
-rw-rw-r--  1 username username       9 Oct 28 09:49  test.txt
-rw-rw-r--  1 username username       5 Oct 28 10:07  .text
```

**Output of `checkrr check`**

_PS: There's Discord spam in the log, though I see no Discord directives in --help_

```
            _|                               _|
   _|_|_|   _|_|_|       _|_|       _|_|_|   _|  _|     _|  _|_|   _|  _|_|
 _|         _|    _|   _|_|_|_|   _|         _|_|       _|_|       _|_|
 _|         _|    _|   _|         _|         _|  _|     _|         _|
   _|_|_|   _|    _|     _|_|_|     _|_|_|   _|    _|   _|         _|
Using config file: /etc/checkrr.yaml
WARN[0000] Sonarr integration not enabled. Files will not be fixed. (if you expected a no-op, this is fine)  startup=true
WARN[0000] Radarr integration not enabled. Files will not be fixed. (if you expected a no-op, this is fine)  startup=true
WARN[0000] Lidarr integration not enabled. Files will not be fixed. (if you expected a no-op, this is fine)  startup=true
INFO[0000] /home/username/testrr/test.mp3  FFProbe=true Format="MP2/3 (MPEG audio layer 2/3)" Type=Audio
INFO[0000] File "/home/username/testrr/test.csv" is not a recongized file type  FFProbe=false Type=Unknown
ERRO[0000] Could not notify Discord
INFO[0000] Couldn't find a target for file "/home/username/testrr/test.csv". File is unknown.  Unknown File=true
INFO[0000] File "/home/username/testrr/test.txt" is not a recongized file type  FFProbe=false Type=Unknown
ERRO[0000] Could not notify Discord
INFO[0000] Couldn't find a target for file "/home/username/testrr/test.txt". File is unknown.  Unknown File=true
+----------------------+-------------+
| Files Checked        |           3 |
| Hash Matches         |           0 |
| Hashes Mismatched    |           0 |
| Submitted to Sonarr  |           0 |
| Submitted to Radarr  |           0 |
| Submitted to Lidarr  |           0 |
| Video Files          |           0 |
| Audio Files          |           1 |
| Text or Other Files  |           0 |
| Unknown Files        |           2 |
| Unknown File Deletes |           0 |
| Elapsed Time         | 48.802153ms |
+----------------------+-------------+
```

**Contents of `badfiles.csv`**

```
/home/username/testrr/test.csv,unknown
/home/username/testrr/test.txt,unknown
```

**Comment by aetaric** [comment]
The discord error is from a line I left in erroneously. 

As for the ignored exts... I'm not seeing the same behavior:

```
checkrr check --config checkrr.yaml

            _|                               _|
   _|_|_|   _|_|_|       _|_|       _|_|_|   _|  _|     _|  _|_|   _|  _|_|
 _|         _|    _|   _|_|_|_|   _|         _|_|       _|_|       _|_|
 _|         _|    _|   _|         _|         _|  _|     _|         _|
   _|_|_|   _|    _|     _|_|_|     _|_|_|   _|    _|   _|         _|
Using config file: checkrr.yaml
INFO[0000] Sonarr Connected.                             startup=true
INFO[0000] Radarr Connected.                             startup=true
INFO[0000] Lidarr Connected.                             startup=true
INFO[0000] Connected to Discord                          startup=true
INFO[0000] /home/aetaric/test/The Devil Wears Prada - Color Decay - 01 - Exhibition.flac  FFProbe=true Format="raw FLAC" Type=Audio
+----------------------+------------+
| Files Checked        |          1 |
| Hash Matches         |          0 |
| Hashes Mismatched    |          0 |
| Submitted to Sonarr  |          0 |
| Submitted to Radarr  |          0 |
| Submitted to Lidarr  |          0 |
| Video Files          |          0 |
| Audio Files          |          1 |
| Text or Other Files  |          0 |
| Unknown Files        |          0 |
| Unknown File Deletes |          0 |
| Elapsed Time         | 90.79613ms |
+----------------------+------------+
```

```yaml
checkpath:
  - ./test
database: checkrr.db
csvfile: "./badfiles.csv"
discordwebhook: "snipped"
notificationtypes:
  - reacquire
  - unknowndetected
  - unknowndeleted
ignoreexts:
  - .txt
  - .csv
  - .DS_Store
```

```
ls -lan test
total 28792
drwxr-xr-x  2 5000 5000     4096 Oct 28 18:57  .
drwx------ 17 5000 5000     4096 Oct 28 19:01  ..
-rw-r--r--  1 5000 5000        0 Oct 28 18:57  .DS_Store
-rw-r--r--  1 5000 5000        0 Oct 28 18:57  test.csv
-rw-r--r--  1 5000 5000        0 Oct 28 18:57  test.txt
-rwxr--r--  1 5000 5000 29478142 Oct 28 18:57 'The Devil Wears Prada - Color Decay - 01 - Exhibition.flac'
```

**Comment by Emstar** [comment]
Wow, weird. Thanks for looking... I will let you know if I figure this out. So odd.

**Comment by Emstar** [comment]
@aetaric I figured it out. You cannot have both `ignoreHidden` and `ignoreExts` configured at the same time. As soon as I commented out `ignoreHidden` I got the expected results from my extension list. You can repro with the config file below and a test directory with appropriate files. Just toggle a comment on the `ignoreHidden: true` config line. 

config file

```
checkpath:
- ./testdir
database: db.db
csvfile: badfiles.csv
# When ignoreHidden is set, ignoreExts is not honored
ignoreHidden: true
ignoreexts:
  - .txt
  - .csv

```

**Comment by aetaric** [comment]
That is... a wild bug, and I can confirm that does it. I'll get a fix for this rolled into 1.2.1.

**Comment by Emstar** [comment]
It was not what I expected, yeah! I almost didn't try. "That would be crazy." 

Oh also, I found that the extension list is case sensitive, it would be sweet to make it insensitive. 

I should get back to work. ;)
